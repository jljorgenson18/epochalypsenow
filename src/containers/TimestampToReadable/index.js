export * from './TimestampToReadable';
export { default } from './TimestampToReadable';
