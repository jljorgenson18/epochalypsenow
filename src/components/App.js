import React, { Component } from 'react';
import { normalize } from 'polished';
import styled, { injectGlobal } from 'styled-components';
import { hot } from 'react-hot-loader';

import TimestampToReadable from './TimestampToReadable';
import ReadableToTimestamp from './ReadableToTimestamp';
import Output from './Output';

injectGlobal`
  ${normalize()}

  @import url('https://fonts.googleapis.com/css?family=Roboto');

  * {
    font-family: 'Roboto', sans-serif;
    margin: 0;
    padding: 0;
  }
`;

const Wrapper = styled.main`
  background-color: #fff2f0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
`;

class App extends Component {
  render() {
    return (
      <Wrapper>
        <h1>Epochalypse Now!</h1>
        <TimestampToReadable />
        <ReadableToTimestamp />
      </Wrapper>
    );
  }
}

export default hot(module)(App);
