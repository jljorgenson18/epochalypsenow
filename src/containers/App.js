import React, { Component } from 'react';
import { normalize } from 'polished';
import styled, { injectGlobal } from 'styled-components';
import { hot } from 'react-hot-loader';

import TimestampToReadable from './TimestampToReadable/index';
import ReadableToTimestamp from './ReadableToTimestamp/index';
import Tabs from '../components/tabs/Tabs';
import Tab from '../components/tabs/Tab';

injectGlobal`
  ${normalize()}

  @import url('https://fonts.googleapis.com/css?family=Roboto');

  * {
    font-family: 'Roboto', sans-serif;
    margin: 0;
    padding: 0;
  }
`;

const Wrapper = styled.main`
  background-color: #fff2f0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
`;

class App extends Component {
  state = {
    activeSection: 'readableToTimestamp'
  };

  handleClick = section => {
    this.setState({ activeSection: section });
  };

  render() {
    const { activeSection } = this.state;
    return (
      <Wrapper>
        <h1>Epochalypse Now!</h1>
        <Tabs>
          <Tab
            section="readableToTimestamp"
            onClick={this.handleClick}
            activeTab={activeSection}>
            Readable To Timestamp
          </Tab>
          <Tab
            section="timestampToReadable"
            onClick={this.handleClick}
            activeTab={activeSection}>
            Timestamp To Readable
          </Tab>
        </Tabs>
        {activeSection === 'readableToTimestamp' ? (
          <ReadableToTimestamp />
        ) : null}
        {activeSection === 'timestampToReadable' ? (
          <TimestampToReadable />
        ) : null}
      </Wrapper>
    );
  }
}

export default hot(module)(App);
