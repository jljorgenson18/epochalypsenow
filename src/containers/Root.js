import React, { Component } from 'react';
// import { IntlProvider, addLocaleData } from 'react-intl';
import { hot } from 'react-hot-loader';
import moment from 'moment-timezone';
import { I18nextProvider } from 'react-i18next';

import App from './App';
import i18n, { initialize as i18nInitialize } from '../i18n/index';

const getLocale = () => {
  return 'fr-CH'; //navigator.language;
};

const loadMomentData = localName => {
  // We already have it if we are english us
  if (localName === 'en-US' || localName === 'en') {
    return Promise.resolve();
  }
  return import(/* webpackChunkName: "moment-locales/[index]" */ `moment/locale/${localName.toLowerCase()}.js`)
    .catch(err =>
      import(/* webpackChunkName: "moment-locales/[index]" */ `moment/locale/${
        localName.split('-')[0]
      }.js`)
    )
    .catch(err => console.error(err));
};

class Root extends Component {
  constructor(props) {
    super(props);
    const locale = getLocale();
    this.state = { i18nInitialized: false, locale };
    this.initializeI18n(locale);
  }
  initializeI18n = locale => {
    return Promise.all([loadMomentData(locale), i18nInitialize(locale)]).then(
      () => {
        moment.locale(locale);
        this.setState({
          i18nInitialized: true
        });
      }
    );
  };
  render() {
    const { i18nInitialized } = this.state;
    if (i18nInitialized) {
      return (
        <I18nextProvider i18n={i18n}>
          <App />
        </I18nextProvider>
      );
    }
    return null;
  }
}

export default hot(module)(Root);
