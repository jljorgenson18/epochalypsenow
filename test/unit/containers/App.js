import React from 'react';
import { expect } from 'chai';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AppComponent } from '~/src/containers/App';

describe('App', () => {
  let mockProps;
  beforeEach(() => {
    mockProps = {
      t: sinon.stub().callsFake(key => key)
    };
  });

  afterEach(() => {
    sinon.restore();
  });

  it('should be here', () => {
    expect(AppComponent).to.be.ok;
  });

  it('should render the app', () => {
    // Arrange

    // Act
    const wrapper = shallow(<AppComponent {...mockProps} />);

    // Assert
    expect(wrapper.find('App__Wrapper')).to.have.length(1);
  });

  it('should match the snapshot', async () => {
    // Arrange

    // Act
    const wrapper = shallow(<AppComponent {...mockProps} />);

    // Assert
    expect(toJson(wrapper)).to.matchSnapshot();
  });

  // Waiting until Enzyme figures their shit out
  it('should update the activeSection when clicking a tab', () => {
    // Arrange
    const wrapper = shallow(<AppComponent {...mockProps} />);

    // Act
    wrapper
      .find('Tabs Tab')
      .at(1)
      .dive()
      .find('button')
      .simulate('click');

    // Assert
    expect(wrapper.state('activeSection')).to.equal('timestampToReadable');
  });
});
