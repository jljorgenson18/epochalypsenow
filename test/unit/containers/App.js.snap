// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<App__Wrapper>
  <GlobalStyleComponent />
  <App__Header>
    <h1>
      Epochalypse Now
    </h1>
    <CurrentTimeStamp />
  </App__Header>
  <App__Calculator>
    <Tabs>
      <Tab
        activeTab="readableToTimestamp"
        onClick={[Function]}
        section="readableToTimestamp"
      >
        Readable To Timestamp
      </Tab>
      <Tab
        activeTab="readableToTimestamp"
        onClick={[Function]}
        section="timestampToReadable"
      >
        Timestamp To Readable
      </Tab>
    </Tabs>
    <App__ActiveSection>
      <ReadableToTimestamp />
    </App__ActiveSection>
  </App__Calculator>
</App__Wrapper>
`;
